service/history/nDCTaskUtil.go

```diff
@@ -83,17 +84,28 @@ func loadMutableStateForTransferTask(
 		metricsClient.Scope(metrics.TransferQueueProcessorScope),
 		logger,
 	)
-	switch err.(type) {
-	case *serviceerror.NotFound:
-		// NotFound error will be ignored by task error handling logic, so log it here
-		// for transfer tasks, mutable state should always be available
-		logger.Error("Transfer Task Processor: workflow mutable state not found, skip.")
-	case *serviceerror.NamespaceNotFound:
-		// NamespaceNotFound error will be ignored by task error handling logic, so log it here
-		// for transfer tasks, namespace should always be available.
-		logger.Error("Transfer Task Processor: namespace not found, skip.")
+	if err != nil {
+		// When standby task executor executes task in active cluster (and vice versa),
+		// mutable state might be already deleted by active task executor and NotFound is a valid case which shouldn't be logged.
+		// Unfortunately, this will also skip logging of actual errors that might happen due to serious bugs,
+		// but these errors, most likely, will happen for other task types too, and will be logged.
```

Comment:
Put a TODO: remove this logic once guaranteed execution and multi-cursor is implemented.